Add an API proposal for browser.readingList()

dlbjames opened the pull request:
## Description
Adds an API proposal for the ReadingList feature in the Chromium browser. Covers the background of the problem, and the initial solution / schema for the API.

Note: The Chromium ReadingList uses the URL of the entries as the key, meaning no duplicate URLs are allowed. This may need to be addressed browser to browser depending on how the feature is implemented.

## Links
Tracking Issue: [1421058](https://bugs.chromium.org/p/chromium/issues/detail?id=1421058)
Feature Request: [1238372](https://bugs.chromium.org/p/chromium/issues/detail?id=1238372)
[Chromium API Proposal Doc](https://docs.google.com/document/d/1f1wW2955nY5Rp3vfS0JGuo1kACOAmsjZupvLYknWKII/edit)

Commit:
Add ReadingList API proposal

oliverdunk commented on proposals/reading_list.md:
````diff
@@ -0,0 +1,178 @@
+```js
+ dictionary ReadingListEntry {
+	// The url of the entry.
+	DOMString url;
````
Could we say a bit more about what we mean by URL here? Can it have query parameters? Can you have two entries with the same hostname and path but a different hash (for example)?

dlbjames replied:
Oh I see now! 

Yes, the URL can have query parameters. Two entries with the same hostname and path but different hashes are also allowed. As long as the string of the url is different even by a single letter. I will update the comment to reflect this. 

Another thing is that the URL must be "valid". For chromium, this means `chrome://` paths cannot be added in addition to any URLs that do not have protocols or hostnames.

dlbjames replied:
Thank you!

dlbjames committed:
Update URL parameter description in all dictionaries

dlbjames committed:
Update reading_list.md

Updates the schema to reflect the current state of implementation on the readingList api

dlbjames committed:
Update reading_list.md

Updated description for url properties with a brief description with what urls are allowed in the reading list.

Changes OnEntryWillBeRemoved -> OnEntryRemoved

dlbjames committed:
Remove supportsPromises from the events

dotproto reviewed (approved):
Looks good to merge
